config/confighttp/compression_test.go

```diff
@@ -104,19 +113,27 @@ func TestHTTPClientCompression(t *testing.T) {
 
 			req, err := http.NewRequest(http.MethodGet, srv.URL, reqBody)
 			require.NoError(t, err, "failed to create request to test handler")
-
+			parts := strings.Split(string(tt.encoding), "/")
+			compressionLevel := configcompression.Level(gzip.BestSpeed)
+			compressionType := configcompression.Type(parts[0])
+			if len(parts) == 2 {
```

Comment:
You can use unmarshalText instead of replicating the code in the test

Reply:
Made this change.